hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/HeFQUINEngineBuilder.java

```diff
@@ -0,0 +1,185 @@
+		this.fedCat = FederationDescriptionReader.readFromModel(fedCatModel);
+		return this;
+	}
+
+	/**
+	 * Sets the engine configuration to be used by the engine.
+	 *
+	 * @param engineConf an engine configuration
+	 * @return this builder instance for method chaining
+	 */
+	public HeFQUINEngineBuilder withEngineConfiguration( final Model engineConf ) {
+		this.engineConf = engineConf;
+		return this;
+	}
+
+	/**
+	 * Sets the engine configuration to be used by the engine.
+	 *
+	 * @param engineConfFile an engine configuration file
+	 * @return this builder instance for method chaining
+	 */
+	public HeFQUINEngineBuilder withEngineConfiguration( final String engineConfFile ) {
+		this.engineConf = RDFDataMgr.loadModel(engineConfFile);
+		return this;
+	}
+
+	/**
+	 * Sets the logical plan printer to be used by the engine.
+	 *
+	 * @param printer a logical plan printer
+	 * @return this builder instance for method chaining
+	 */
+	public HeFQUINEngineBuilder withLogicalPlanPrinter( final LogicalPlanPrinter printer ) {
+		this.lplanPrinter = printer;
+		return this;
+	}
+
+	/**
+	 * Sets the physical plan printer to be used by the engine.
+	 *
+	 * @param printer a physical plan printer
+	 * @return this builder instance for method chaining
+	 */
+	public HeFQUINEngineBuilder withPhysicalPlanPrinter( final PhysicalPlanPrinter printer ) {
+		this.pplanPrinter = printer;
+		return this;
+	}
+
+	/**
+	 * Sets the source assignment printer to be used by the engine.
+	 *
+	 * @param ptiner a source assignment printer
+	 * @return this builder instance for method chaining
+	 */
+	public HeFQUINEngineBuilder withSourceAssignmentPrinter( final LogicalPlanPrinter printer ) {
+		this.srcasgPrinter = printer;
+		return this;
+	}
+
+	/**
+	 * Sets whether the engine should skip execution after query planning.
+	 *
+	 * @param skip whether to skip query execution
+	 * @return this builder instance for method chaining
+	 */
+	public HeFQUINEngineBuilder withSkipExecution( final boolean skip ) {
+		this.skipExecution = skip;
+		return this;
+	}
+
+	/**
+	 * Sets whether the engine should treat the current run as part of an experiment.
+	 *
+	 * @param isExperimentRun whether this is an experimental run
+	 * @return this builder instance for method chaining
+	 */
+	public HeFQUINEngineBuilder withExperimentRun(final boolean isExperimentRun) {
+		this.isExperimentRun = isExperimentRun;
+		return this;
+	}
+
+	/**
+	 * Builds and initializes a {@link HeFQUINEngineAndContext} instance using the
+	 * parameters configured via this builder.
+	 *
+	 * The federation catalog must be provided via {@code withFederationCatalog},
+	 * default values will be used for all other components unless specified explicitly.
+	 *
+	 * @return an initialized {@link HeFQUINEngine} instance
+	 * @throws IllegalStateException if the federation catalog has not been set
+	 */
+	public HeFQUINEngine build() {
+		assert fedCat != null;
```

Comment:
I believe the `throws`- comment is wrong. I don't think that `assert` throws an `IllegalStateException`.

Please replace the `assert` and do throw an `IllegalStateException`.

Reply:
It seems you forgot to address this comment.

Reply:
Oops, must've misclicked. Will address it.